### PR TITLE
StatsFactory: Initialize stats only once

### PR DESCRIPTION
**Pull request by REASY**
Resolving https://github.com/LBNL-UCB-STI/beam/issues/1220

---
This change is [Reviewable](https://reviewable.io/reviews/lbnl-ucb-sti/beam/1221)

### CODE REVIEW

File: src/main/java/beam/analysis/StatsFactory.java [hunk 1]
```diff
@@ -42,8 +42,11 @@ public StatsFactory(BeamServices services) {
     }
 
     public BeamAnalysis getAnalysis(StatsType statsType) {
-        BeamAnalysis stats = beamStatsMap.getOrDefault(statsType, createStats(statsType));
-        beamStatsMap.putIfAbsent(statsType, stats);
+        BeamAnalysis stats = beamStatsMap.get(statsType);
```
Comment (JustinPihony) [inline src/main/java/beam/analysis/StatsFactory.java]: How about using `computeIfAbsent` and turning it into a one liner :) ?
```
return beamStatsMap.computeIfAbsent(statsType, StatsFactory::createStats);
```

*The method reference might need tweaked as I didnt run this against a compiler

**Comment by colinsheppard** [comment]
test!

**Comment by REASY** [comment]
Test!

**Comment by REASY** [comment]
Test!

**Pull request merged**